src/v/cloud_storage/CMakeLists.txt

```diff
@@ -4,6 +4,7 @@ v_cc_library(
   SRCS
     manifest.cc
     remote.cc
+    cache_service.cc
```

Comment:
```
archival: add cloud_storage cache service

The cloud storage cache will be used by cloud_storage to cache manifests
and segments from remote storage.
```

it would be nice if the commit message included a bit more detail about the caching service. a few sentences describing the high-level architecture and/or the salient components is probably enough.

Reply:
Added new commit message.